spec/models/community_users.rb

```diff
@@ -0,0 +1,84 @@
+require 'spec_helper'
+
+describe CommunityUsers do
+  let(:student) { create(:student) }
+  let(:student_looking_for_coach) { create(:student, :interested_in_coaching) }
+  let(:coach) { create(:coach, :available) }
+  describe '#all' do
+    before do
+      @params = {
+          role: '',
+          interest: '',
+          availability: '',
+          search: '',
+          location: '',
+          page: 1,
+          sort: '',
+          direction: 'asc'
+        }
+
+      @users = User.ordered(@params[:sort], @params[:direction])
+                   .group('users.id')
+                   .with_all_associations_joined
+
+      @community = CommunityUsers.new(@params, @users)
+    end
+
+    context 'return filtered users' do
+      describe 'with params presence' do
```

Comment:
This should be a `context` instead. On a technical level, this does of course make no difference, but things beginning with "with" and "when" are usually `context` blocks and make reading easier.